src/remediations/fifi_2.js

```diff
@@ -217,3 +219,112 @@ async function storePlaybookRun (playbook_run_id, remediation, username) {
+                COUNT(CASE WHEN dr.status IN ('failure', 'timeout') THEN 1 END)::int as failed_runs,
+                COUNT(CASE WHEN dr.status IN ('pending', 'running') THEN 1 END)::int as incomplete_runs
+            FROM playbook_runs pr
+            LEFT JOIN dispatcher_runs dr ON pr.id = dr.remediations_run_id
+            WHERE pr.id IN (:playbookRunIds)
+            GROUP BY pr.id
+        `;
+
+        const playbookRunCounts = await db.s.query(sql, {
+            replacements: { playbookRunIds },
+            type: db.s.QueryTypes.SELECT
+        });
+
+        const needsStatusSync = [];
+        const needsBackfill = [];
+
+        for (const runCounts of playbookRunCounts) {
+            if (runCounts.total_dispatcher_runs === 0) {
+                // No dispatcher_runs exist, needs backfill
+                needsBackfill.push(runCounts.playbook_run_id);
+            } else if (runCounts.incomplete_runs > 0 && runCounts.failed_runs === 0) {
+                // Has pending/running, needs update
+                needsStatusSync.push(runCounts.playbook_run_id);
+            }
+        }
+
+        const runsToSync = [...needsStatusSync, ...needsBackfill];
+
+        if (runsToSync.length > 0) {
+            // Limit concurrent API calls
+            const limit = pLimit(10);
+            
+            await Promise.all(runsToSync.map(playbookRunId => limit(async () => {
+                try {
+                    const filter = {
+                        filter: { 
+                            service: 'remediations',
+                            labels: { 'playbook-run': playbookRunId } 
+                        }
+                    };
+                    const fields = {
+                        fields: { data: ['id', 'status'] }
+                    };
+                    
+                    const dispatcherResponse = await dispatcher.fetchPlaybookRuns(filter, fields);
+                    
+                    if (dispatcherResponse?.data) {
+                        if (needsBackfill.includes(playbookRunId)) {
+                            // Create new dispatcher_runs entries (backfill)
+                            const newRuns = dispatcherResponse.data.map(pdRun => ({
+                                dispatcher_run_id: pdRun.id,
+                                remediations_run_id: playbookRunId,
+                                status: pdRun.status,
+                                created_at: new Date(),
+                                updated_at: new Date(),
+                                pd_response_code: null
+                            }));
+                            
+                            await queries.insertDispatcherRuns(newRuns);
+                        } else {
+                            // Update existing dispatcher_runs entries
+                            for (const pdRun of dispatcherResponse.data) {
+                                await db.dispatcher_runs.update(
```

Comment:
Don't do this - you're calling `queries.insertDispatcherRuns()` above to add new runs; add an `updateDispatcherRuns()` function to `queries` and call that instead of calling the model object directly.